measures/BuildResidentialModel/measure.rb

```diff
@@ -106,37 +118,112 @@ def run(model, runner, user_arguments)
 
       # HPXMLtoOpenStudio
       measure_subdir = 'HPXMLtoOpenStudio'
+
       full_measure_path = File.join(measures_dir, measure_subdir, 'measure.rb')
       check_file_exists(full_measure_path, runner)
-      measure = get_measure_instance(full_measure_path)
 
-      # Fill the measure args hash with default values
       measure_args = {}
-
       measures[measure_subdir] = []
-      measure_args[:hpxml_path] = File.expand_path('../in.xml')
+      measure_args[:hpxml_path] = File.expand_path('../out.xml')
       measure_args[:weather_dir] = File.expand_path('../../../../weather')
       measure_args[:output_dir] = File.expand_path('..')
-      measure_args[:debug] = false
+      measure_args[:debug] = true
       measure_args = Hash[measure_args.collect{ |k, v| [k.to_s, v] }]
       measures[measure_subdir] << measure_args
-
-      if not apply_measures(measures_dir, measures, runner, unit_model, workflow_json, unit_name, true)
+      if not apply_measures(measures_dir, measures, runner, unit_model, workflow_json, 'out.osw', true)
         return false
       end
 
+      unit_dir = File.expand_path("../unit #{num_unit}")
+      Dir.mkdir(unit_dir)
+      FileUtils.cp(File.expand_path('../out.osw'), unit_dir)
+      FileUtils.cp(File.expand_path('../out.xml'), unit_dir)
+      FileUtils.cp(File.expand_path('../in.xml'), unit_dir)
+      FileUtils.cp(File.expand_path('../in.osm'), unit_dir)
+
+      # create building unit object to assign to spaces
+      building_unit = OpenStudio::Model::BuildingUnit.new(unit_model)
+      building_unit.setName("building_unit_#{num_unit}")
+
+      # save modified copy of model for use with merge
+      unit_model.getSpaces.sort.each do |space|
+        space.setYOrigin(60 * (num_unit -1)) # meters
+        space.setBuildingUnit(building_unit)
+      end
+
+      # prefix all objects with name using unit number. May be cleaner if source models are setup with unique names
+      unit_model.objects.each do |model_object|
+        next if model_object.name.nil?
+        model_object.setName("unit_#{num_unit} #{model_object.name.to_s}")
+      end
+
+      moodified_unit_path = File.expand_path("../unit #{num_unit}/modified_unit.osm")
+      unit_model.save(moodified_unit_path, true)
+
+      # passing modified copy into array, can move earlier if we don't want the modified copy
       unit_models << unit_model
+
+
+      # run merge merge_spaces_from_external_file to add this unit to original model
+      merge_measures_dir = nil
+      osw_measure_paths = runner.workflow.measurePaths
```

Comment:
Could be easier to just store (a custom version of) `merge_spaces_from_external_file` alongside BuildResidentialModel.